specs/protocol/flashblocks.md

```diff
@@ -0,0 +1,1029 @@
+
+However, an important implementation detail is that these post-block changes must be reverted before beginning execution for the next flashblock. This reversion is necessary because the post-block operations should only be applied once per actual L2 block, not cumulatively for each flashblock. Failing to revert these changes would lead to their repeated application across multiple flashblocks, potentially creating invalid cumulative state and ultimately an invalid final block.
+
+### Construction Steps
+
+After handling the mandatory sequencer transactions in the initial Flashblock, the External Block Builder proceeds with constructing subsequent Flashblocks by following these steps for each interval:
+
+1.	**Transaction Selection**
+
+- Retrieve transactions from local or external mempool:
+- Prioritize and sort transactions based on predefined sequencing policies, such as priority ordering or by MEV paid.
+
+2.	**Transaction Execution**
+
+- Sequentially execute selected transactions against a state snapshot derived from the current execution payload base (ExecutionPayloadBaseV1) or the last validated flashblock
+- Apply the transaction inclusion heuristics described earlier to determine when to stop including transactions
+- After transaction execution completes, apply all post-block execution rules as described in the Post-Block Execution Rules section
+
+3.	**Flashblock Payload Assembly**
+
+- After transaction execution, compute and record the following execution state updates:
+    - `state_root`: The new post-execution state root resulting from the executed transactions.
+    - `receipts_root`: The receipts trie root derived from execution outcomes.
+    - `logs_bloom`: Aggregated logs bloom from all emitted transaction logs within this flashblock.
+    - `gas_used`: Total gas consumed by executed transactions.
+    - `transactions`: Serialized transaction payloads included within the flashblock.
+    - `withdrawals` (if applicable): Withdrawals executed during the current flashblock interval (as per OP Stack withdrawal specification).
+    - `block_hash`: Computed block hash uniquely identifying this flashblock execution state.
+    
+    Note that each flashblock builds upon the state of all previous flashblocks, with these fields reflecting the cumulative state after applying the new transactions in this particular flashblock.
+    
+- Encapsulate these computed updates into `ExecutionPayloadFlashblockResultV1`.
+
+5.	**Flashblock Indexing and Metadata**
+
+- Assign a monotonically incremented `index` to the newly constructed Flashblock payload.
+- Compute the SSZ hash of the previous Flashblock and assign it as the `parent_flash_hash` (for the first Flashblock with index 0, this field is empty)
+
+6.	**Flashblock Delivery**
+
+- Package the `index`, `payload_id`,  `ExecutionPayloadFlashblockDeltaV1`, and metadata into a `FlashblocksPayloadV1` payload.
```

Comment:
NIT: We don't have the structure for this struct`ExecutionPayloadFlashblockDeltaV1`can we add a link of add them on the struct list above?